runtime/vm/JVMImage.cpp

```diff
@@ -74,30 +125,69 @@ JVMImage::freeSubAllocatedMemory(void* address)
 }
 
 void
-JVMImage::reallocateImageMemory(UDATA size)
+JVMImage::readImageFromFile()
 {
-	return;
+    OMRPORT_ACCESS_FROM_OMRPORT(getPortLibrary());
+
+    char imageBuffer[_imageSize];
+    memset(imageBuffer, 0, sizeof(imageBuffer));
+
+    intptr_t fileDescriptor = omrfile_open(_dumpFileName, EsOpenRead, 0666);
+    if (fileDescriptor == -1) {
+        // Failure to open file
+    }
+
+    intptr_t bytesRead = omrfile_read(fileDescriptor, imageBuffer, sizeof(imageBuffer));
+    if ((bytesRead == 0) || (bytesRead == -1)) {
+        // Failure to read the image
+    }
+
+    if (omrfile_close(fileDescriptor) != 0) {
+        // Failure to close
```

Comment:
it might be worth exploring an enum for error codes or something to handle all the read write error code. Not important for this PR